x-pack/legacy/plugins/reporting/export_types/common/lib/screenshots/index.ts

```diff
@@ -25,116 +14,57 @@ import { getNumberOfItems } from './get_number_of_items';
 import { waitForElementsToBeInDOM } from './wait_for_dom_elements';
 import { getTimeRange } from './get_time_range';
 import { getElementPositionAndAttributes } from './get_element_position_data';
-import { getScreenshots } from './get_screenshots';
+import { captureScreenshotData } from './capture_screenshots';
 
-export function screenshotsObservableFactory(server: KbnServer) {
-  const browserDriverFactory: HeadlessChromiumDriverFactory = server.plugins.reporting.browserDriverFactory; // prettier-ignore
+export function screenshotsFactory(server: KbnServer) {
   const config = server.config();
   const captureConfig = config.get('xpack.reporting.capture');
 
-  return function screenshotsObservable({
+  return async function getScreenshots({
+    browser,
     logger,
     url,
     conditionalHeaders,
     layout,
     browserTimezone,
-  }: ScreenshotObservableOpts): Rx.Observable<void> {
-    const create$ = browserDriverFactory.create({
-      viewport: layout.getBrowserViewport(),
-      browserTimezone,
-    });
+  }: ScreenshotObservableOpts): Promise<{
+    timeRange: TimeRange | null;
+    screenshots: Screenshot[];
+  }> {
+    await openUrl(browser, url, conditionalHeaders, logger);
+
+    const successSelectors = `${layout.selectors.renderComplete},[${layout.selectors.itemsCountAttribute}]`;
+    await Promise.race([
+      browser.waitForSelector(successSelectors, {}, logger), // finds DOM attributes for Kibana embeddables
+      checkForToastMessage(browser, layout, logger), // if this wins the race, there's an error on the page
+    ]);
+
```

Comment:
This is much more readable, and less complex, than doing the mergemap chain

Reply:
it also seems to not work nearly as well :(